rust/worker/src/segment/types.rs

```diff
@@ -81,6 +168,25 @@ pub(crate) struct MaterializedLogRecord<'referred_data> {
     pub(crate) final_embedding: Option<&'referred_data [f32]>,
 }
 
+pub(crate) struct MetadataDelta<'referred_data> {
```

Comment:
this types file is turning into a junk drawer. Can we be more empathetic to future readers with the placement of these types / new files?